src/backend/renderer/output/mod.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::VecDeque;
+
+use indexmap::IndexMap;
+
+use crate::{
+    backend::renderer::Frame,
+    utils::{Physical, Rectangle, Scale, Size},
+    wayland::output::Output,
+};
+
+use self::element::{Id, RenderElement};
+
+use super::{ImportAll, Renderer, Texture};
+
+pub mod element;
+
+#[derive(Debug, Clone, Copy)]
+struct ElementState {
+    last_commit: usize,
+    last_geometry: Rectangle<i32, Physical>,
+    last_z_index: usize,
+}
+
+#[derive(Debug)]
+struct OutputRenderState {
+    size: Option<Size<i32, Physical>>,
+    elements: IndexMap<Id, ElementState>,
+}
+
+/// Rendering for a single output
+#[derive(Debug)]
+pub struct OutputRender {
+    output: Output,
+    last_states: VecDeque<OutputRenderState>,
+    empty_state: OutputRenderState,
+}
+
+impl OutputRender {
+    /// Initialize a new [`OutputRender`]
+    pub fn new(output: &Output) -> Self {
```

Comment:
At some point I would like to make this generic.

We essentially only need size, scale and transform from the output.
Storing the output has the nice advantage, that this updates with said output, but it should be possible for users to create this struct from a fixed size/scale/transform.

Maybe we should also rename this to `DamageTrackedRenderer` or something.

Reply:
Do we really have to store the size/scale/transform combo on creation? We could also make them part of the `render_output` function and only store them in the last state for checking if they changed. Then it would be clear from the API side and it would not be possible to forget updating them.

Reply:
I don't want to store them unconditionally. I just want to have the option to either initialize the renderer with an output (and have them update automatically) or initialize it statically.